speech-color-changer/script.js

```diff
@@ -1,14 +1,18 @@
 var SpeechRecognition = SpeechRecognition || webkitSpeechRecognition
-var SpeechGrammarList = SpeechGrammarList || webkitSpeechGrammarList
+var SpeechGrammarList = SpeechGrammarList || window.webkitSpeechGrammarList
```

Comment:
This looks like a mistake to me. Did you meant to fix the line above?
```
var SpeechRecognition = SpeechRecognition || webkitSpeechRecognition || window.webkitSpeechRecognition
```

Reply:
no, but we could fix it there too. Safari is missing `webkitSpeechGrammarList` and so throws an error unless you check the property on window. It wasn't missing `webkitSpeechRecognition` in the version I tested so I didn't see an error there.